### DISCUSSION TITLE
tutorials/voyage-part2/02-monolith-graph-setup

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#68] [course feedback]
# tutorials/voyage-part2/02-monolith-graph-setup

https://www.apollographql.com/tutorials/voyage-part2/02-monolith-graph-setup

**Comment by AlexD332** [comment]
Hi, 

I can't publish Local Schema as my modal dialog doesn't have such tab. 
https://prnt.sc/DklM0vyY0aGB
What should I do?

Thanks,
Alex.

**Reply by lizhennessy** [reply]
Hey @AlexD332! Happy to help troubleshoot. For this course we'll need to be on the Enterprise plan, as we're creating a different type of graph than the one that comes default through the onboarding flow. Are you on an Enterprise plan currently? (Were you able to set the graph architecture to "monolith" as detailed in an earlier step?) Thanks!

**Reply by AlexD332** [reply]
Hi @lizhennessy, "(Were you able to set the graph architecture to "monolith" as detailed in an earlier step?)"
yeah, that what I missed I think the screen on that step need to be modal only.
Thanks anyway

**Reply by Zain-ul-din** [reply]
I want to share a quick trick, if you are not working at a company and don't have access to enterprise plan you can still get it for free by providing your student mail.

**Reply by nabeff** [reply]
Hello, is Part 2 and 3 without videos ?

**Reply by lizhennessy** [reply]
Hi there @nabeff , unfortunately that's right, no videos for part 2 and 3 of this series. However, if you're looking to get up and running with federation even faster, we've just released [a new course on Apollo Connectors](https://www.apollographql.com/tutorials/connectors-intro-rest) which lets you skip the steps of writing custom subgraph servers, and instead do all your API orchestration from the schema (This also makes it helpful when deconstructing a monolith!). This course comes with videos and code challenges to test what you've learned along the way. :)

**Reply by Gkodkod** [reply]
@lizhennessy I need access to this course to complete my Professional certification and am not on an enterprise plan yet.  Can you assist me ?

**Comment by Yasir5247** [comment]
I dont understand one thing. imagine i have a big graphql monolith api. and i'm trying to convert into federation. now this api is made from nexus-graphql which means the schema is auto generated and its code first. now i want slowly migrate into federation. the trouble for me is i don't know how split schema and combine it into one.

**Reply by mabuyo** [reply]
Hey there @Yasir5247! The rest of the course walks through the process of splitting the schema into a separate subgraph, then using federation to compose into one supergraph schema. We don't have a code-first example but I imagine that instead of taking the types and fields in the schema, you would split up resolvers in your code instead.

**Reply by Gkodkod** [reply]
Hey there @maybuyo !!!  Thanks for all you classes.  I need access to this course to complete my professional certification and am not on an enterprise plan yet.  Can you assist me ?

**Reply by bignimbus** [reply]
Hi @Gkodkod, I can assist here. You should be able to access this course without an enterprise plan, are you seeing something different on your end?

**Reply by Gkodkod** [reply]
Hi @bignimbus, That would be awesome and thanks in advance !!!

Take a look above... Same issue as AlexD332 on Sep 10, 2023 
I can't publish Local Schema as my modal dialog doesn't have such tab.

@lizhennessy Mar 17, 2025 answered that need an enterprise account and I already took the other course she offered.  But need this course to complete the next stage.  Someone mentioned above to get a student account, but maybe you can resolve this in a much easier way.

I also need this course for the code and topics to move to next stage.

**Reply by Gkodkod** [reply]
Hi @bignimbus, above it says "create a new graph.  We'll publish our schemas to the graph we create here, so give it a helpful name. Change the Graph Architecture to Monolith. Then click Next to move to the next step."

I do not see in my account "Graph Architecture" = Monolith, instead it says Monograph.  Is that the same as monolith ?

**Reply by Gkodkod** [reply]
Hi @bignimbus, I got it !!!  

You guys got to update this tutorial's pics above and instructions.  
1 .It is a monograph.
2. You got to use Rover to publish rather than on Apollo GraphOS Studio 

Thx :)

**Reply by bignimbus** [reply]
Glad you're unblocked, adding this to our backlog!

**Comment by kid1412621** [comment]
now, it's monograph rather than monolith

**Comment by AdamMcCormickOMF** [comment]
I'm definitely on an Enterprise plan and selected the Monograph option, but there's no modal to upload a schema, just a rover command
